daemon/info_unix.go

```diff
@@ -122,10 +126,13 @@ func (daemon *Daemon) fillPlatformInfo(v *types.Info, sysInfo *sysinfo.SysInfo)
 			v.Warnings = append(v.Warnings, "WARNING: Support for cgroup v2 is experimental")
 		}
 		// TODO add fields for these options in types.Info
-		if !sysInfo.BlkioWeight {
+		if !sysInfo.BlkioWeight && v.CgroupVersion == "2" {
+			// blkio weight is not available on cgroup v1 since kernel 5.0.
+			// Warning is not printed on cgroup v1, because there is no action user can take.
+			// On cgroup v2, blkio weight is implemented using io.weight
 			v.Warnings = append(v.Warnings, "WARNING: No blkio weight support")
```

Comment:
Do we need to update the warnings to match the new cgroup? (ok for a follow-up)

Reply:
updated